comm-msg.go

```diff
@@ -89,26 +91,39 @@ func processPayload(b []byte, cmd int, Key []byte) {
 		return
 	} // empty
 	var data []byte
-	if b[0] == byte('{') {
+	if b[0] == byte('{') && version == Version_3_1{
 		//  Message in clear text
 		data = b
 	} else {
-		encrypted := false
-		if len(b) > (len(Version) + 16) {
+		hasVersion := false
+		if len(b) > (len(version) + 16) {
 			// Check if message starts with version number
-			encrypted = true
-			for i, vb := range Version {
-				encrypted = encrypted && b[i] == byte(vb)
+			hasVersion = true
+			for i, vb := range version {
+				hasVersion = hasVersion && b[i] == byte(vb)
 			}
 		}
+		encrypted := true
+		if version == Version_3_1 && !hasVersion {
+			encrypted = false
+		}
 		if !encrypted {
 			fmt.Println("\tClear text", string(b))
 			return
 		}
 		emark = "Encrypted"
 		var err error
 		if len(Key) > 0 {
-			data, err = aesDecrypt(b[len(Version)+16:], Key) // ignore signature
+			if version == Version_3_1 {
+				// Should this be 12?
+				data, err = aesDecrypt(b[len(version)+16:], Key, version) // ignore signature
```

Comment:
Not sure if the offset should be changed to 12 here since the return code is now skipped?